// redux/actions/auth.js
import * as types from 'redux/constants/types/auth';

export function fetchUserPost(user) {

    return (dispatch) => {
        const url = 'https://seryozhabaleyko.auth0.com/api/v2/users';

        const options = {
            method: 'GET',
            mode: 'no-cors',
            headers: {
                'Content-Type': 'application/json',
                Accept: 'application/json',
            },
            // body: JSON.stringify({ user }),
        };

        return fetch(url, options)
            .then(response => response.json())
            .then((data) => {
                if (data.message) {
                    console.error(data.message);
                } else {
                    localStorage.setItem('token', data.token);
                    dispatch(loginUser(data.user));
                }
            });
    };
};

const loginUser = (user) => ({
    type: 'LOGIN_USER',
    payload: user,
});

export const setUsername = (phone) => ({
    type: types.AUTH_CHANGE_USERNAME,
    payload: phone,
});

export const setValidUsername = (bool) => ({
    type: types.AUTH_CHANGE_VALID_USERNAME,
    payload: bool,
});

export const setPassword = (password) => ({
    type: types.AUTH_CHANGE_PASSWORD,
    payload: password,
});

export const setValidPassword = (bool) => ({
    type: types.AUTH_CHANGE_VALID_PASSWORD,
    payload: bool,
});

export const setIsAuthValue = (bool) => ({
    type: types.AUTH_SET_IS_AUTH_VALUE,
    payload: bool,
});
