// redux/actions/profile.js
import types from 'redux/constants/types/auth';

const fetchUser = (token) => new Promise((resolve, reject) => {
    resolve({
        username: 'admin',
        password: '12345',
        avatar: 'https://robohash.org/imanewuser.png',
        bio: 'A new user to the app.',
        token: 'qweqweqweqweqweqweqweqweqweqweqweqweqweqwe',
    });
});

function getProfileFetch() {
    return async (dispatch) => {
        const token = localStorage.token;

        if (token) {
            dispatch({
                type: types.request,
            });

            try {
                const user = await fetchUser(token);

                localStorage.setItem('token', user.token);

                dispatch({
                    type: types.success,
                    payload: user,
                });
            } catch (error) {
                dispatch({
                    type: types.failure,
                    payload: error,
                    error: true,
                });
            }
        }
    };
}

export default getProfileFetch;
